File: Python/SortingAlgorithms/SelectionSort.py
# python program to implement selection sort
def selectionSort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_index = i
        
        for j in range(i + 1, n):
            if (arr[j] < arr[min_index]):
                min_index = j
                
        arr[i], arr[min_index] = arr[min_index], arr[i]
        

def printArray(arr):
    for val in arr:
        print(val, end=" ")
    print()

if __name__ == "__main__":
    arr = [64, 25, 12, 22, 11]
    print("Original array")
    printArray(arr)
    
    print("Sorted array")
    selectionSort(arr)
    printArray(arr)

File: Javascript/SortingAlgorithms/SelectionSort.js
function selectionSort(arr){
    let n = arr.length

    for (let i = 0; i < n; i++) {
        min_index = i

        for (let j=i+1; j < n; j++) {
            if (arr[j] < arr[min_index]) {
                min_index = j
            }
        }
      
        let temp = arr[i]
        arr[i] = arr[min_index]
        arr[min_index] = temp
    }

}

function printArray(arr) {
    for (let val of arr) {
        process.stdout.write(val + " ")
    }
    console.log();
}

// Driver function 
const arr = [64, 25, 12, 22, 11];

console.log("Original Array");
printArray(arr)

console.log("Sorted Array");
selectionSort(arr)
printArray(arr)

